### PR TITLE
KBC-2410 branch metadata final

### PR DESCRIPTION
**Pull request by martinjunger**
Jira: KBC-2410
Connection: https://github.com/keboola/connection/pull/3601

Before asking for review make sure that:

- [x] New client method(s) has tests
- [x] Apiary file is updated
- [x] You declared if there is a BC break or not (will affect next release of a client)

---

### CODE REVIEW

**Comment by tomasfejfar** [comment]
👀

File: tests/Common/BranchMetadataTest.php [hunk 1]
```diff
@@ -227,34 +232,135 @@ public function testDeleteMetadataRestrictionForReadOnlyUser()
...
+        $branchClient = $this->clientProvider->getDevBranchClient();
+        $branchMdClient = new DevBranchesMetadata($branchClient);
+
+        // list copied metadata in branch
+        $branchMetadata = $branchMdClient->listBranchMetadata();
+        $this->assertCount(2, $branchMetadata);
+        $this->assertMetadataEquals(self::TEST_METADATA[0], $branchMetadata[0]);
+        $this->assertGreaterThan($metadata[0]['timestamp'], $branchMetadata[0]['timestamp']);
+        $this->assertMetadataEquals(self::TEST_METADATA[1], $branchMetadata[1]);
+        $this->assertNotSame($metadata[0]['timestamp'], $branchMetadata[0]['timestamp']);
```
Comment (tomasfejfar) [inline tests/Common/BranchMetadataTest.php:300]: Tohle je IMHO špatné chování. Kopírování branche by mělo být 1:1, tj. timestamp by měl být stejný.

File: tests/Common/BranchMetadataTest.php [hunk 2]
```diff
@@ -227,34 +232,135 @@ public function testDeleteMetadataRestrictionForReadOnlyUser()
...
+                'value' => 'Branch value',
+            ],
+        ];
+        $branchMetadata = $branchMdClient->addBranchMetadata($BRANCH_METADATA);
+        $this->assertCount(3, $branchMetadata);
+        $this->assertMetadataEquals(self::TEST_METADATA[0], $branchMetadata[0]);
+        $this->assertMetadataEquals(self::TEST_METADATA[1], $branchMetadata[1]);
+        $this->assertMetadataEquals($BRANCH_METADATA[0], $branchMetadata[2]);
+
+        // delete some metadata from default
```
Comment (tomasfejfar) [inline tests/Common/BranchMetadataTest.php:342]:
```suggestion
        // delete some metadata from branch
```

**Review by tomasfejfar — CHANGES_REQUESTED** [review]

File: tests/Common/BranchMetadataTest.php (hunk 1 above)
Comment (martinjunger) [reply tests/Common/BranchMetadataTest.php:300]: viz https://github.com/keboola/connection/pull/3601#discussion_r810938176

File: tests/Common/BranchMetadataTest.php (hunk 2 above)
Comment (martinjunger) [reply tests/Common/BranchMetadataTest.php]: upraveno, dik

**Review by fejjo — APPROVED** [review]
🙆‍♂️

**Pull request merged**